src/utils/useFiltered_SortedPeople.ts

```diff
@@ -0,0 +1,65 @@
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+
+export const useFilteredSortedPeople = (
+  people: Person[],
+): {
+  sortedPeople: Person[];
+  order: string | null;
+  typeSort: string | null;
+} => {
+  const [searchParams] = useSearchParams();
+  const selectedCenturies = searchParams.getAll('centuries').map(item => +item);
+  const selectedSex = searchParams.get('sex');
+  const order = searchParams.get('order');
+  const typeSort = searchParams.get('sort');
+
+  const matchYear = (year: number) => {
+    if (!selectedCenturies.length) {
+      return true;
+    }
+
+    const minYear = Math.min(...selectedCenturies) * 100 - 100;
+    const maxYear = Math.max(...selectedCenturies) * 100;
+
+    return year >= minYear && year < maxYear;
+  };
+
+  const filteredPeople = people
+    .filter(person => {
+      const query = searchParams.get('query')?.toLowerCase() || '';
+
+      return (
+        person.name.toLowerCase().includes(query) ||
+        person.motherName?.toLowerCase().includes(query) ||
+        person.fatherName?.toLowerCase().includes(query)
+      );
+    })
+    .filter(person => matchYear(person.born))
+    .filter(person => {
```

Comment:
why are u using 3 `filter` methods? U can do only 1, in the middle of which callback u will create 3 conditions
Do it